synapse/config/room_directory.py

```diff
@@ -23,70 +23,121 @@ def read_config(self, config):
         alias_creation_rules = config["alias_creation_rules"]
 
         self._alias_creation_rules = [
-            _AliasRule(rule)
+            _RoomDirectoryRule("alias_creation_rules", rule)
             for rule in alias_creation_rules
         ]
 
+        room_list_publication_rules = config["room_list_publication_rules"]
+
+        self._room_list_publication_rules = [
+            _RoomDirectoryRule("room_list_publication_rules", rule)
+            for rule in room_list_publication_rules
+        ]
+
     def default_config(self, config_dir_path, server_name, **kwargs):
         return """
         # The `alias_creation` option controls who's allowed to create aliases
         # on this server.
         #
         # The format of this option is a list of rules that contain globs that
-        # match against user_id and the new alias (fully qualified with server
-        # name). The action in the first rule that matches is taken, which can
-        # currently either be "allow" or "deny".
+        # match against user_id, room_id and the new alias (fully qualified with
+        # server name). The action in the first rule that matches is taken,
+        # which can currently either be "allow" or "deny".
+        #
+        # Missing user_id/room_id/alias fields default to "*".
         #
         # If no rules match the request is denied.
         alias_creation_rules:
             - user_id: "*"
-              alias: "*"
+              alias: "*"      # This matches alias being created
+              room_id: "*"
+              action: allow
+
+        # The `room_list_publication_rules` option control who and what can be
+        # published in the public room list.
+        #
+        # The format of this option is the same as that for
+        # `alias_creation_rules`
```

Comment:
does this match on all aliases (using the `m.room.aliases` event ?) or just local aliases?